randovania/data/json_data/prime3.txt

```diff
@@ -1641,24 +1933,51 @@ Asset id: 2580753232084104741
 
 > Room Center; Heals? False
   > Dock to Burrow
-      Ice Missile
+      After Hall of the Golems - Top Golem
   > Dock to Hangar Bay Hall
       Trivial
   > Dock to Colossus Vista
-      Plasma Beam
+      After Hall of the Golems - Left Golem
   > Pickup (Missile Expansion)
-      Grapple Voltage and Spider Ball and Boost Ball
+      Spider Ball and Boost Ball and After Hall of the Golems - Right Golem
+  > Event - Right Golem
+      Grapple Voltage
+  > Event - Left Golem
+      Plasma Beam
+  > Event - Top Golem
+      Ice Missile
+
+> Event - Right Golem; Heals? False
+  * Event Hall of the Golems - Right Golem
+  > Room Center
+      Trivial
+
+> Event - Left Golem; Heals? False
+  * Event Hall of the Golems - Left Golem
+  > Room Center
+      Trivial
+
+> Event - Top Golem; Heals? False
+  * Event Hall of the Golems - Top Golem
+  > Room Center
+      Trivial
 
 ----------------
 South Jungle Court
 Asset id: 8404374826148145183
 > Dock to Generator Hall South; Heals? False
   * Normal Door to Generator Hall South/Dock to South Jungle Court
   > Dock to South Jungle Hall
-      Trivial
+      After South Jungle Court Turret
+  > Event - Destroy Turret
+      Grapple Lasso
 
 > Dock to South Jungle Hall; Heals? False
   * Normal Door to South Jungle Hall/Dock to South Jungle Court
+  > Dock to Generator Hall South
+      After South Jungle Court Turret
+
+> Event - Destroy Turret; Heals? False
   > Dock to Generator Hall South
       Trivial
 
```

Comment:
For Thorn Jungle Airdock, add node called "Samus Ship", with "Heals?" checked, and have it connect to the door as "Trivial"